rewrite-kotlin/src/main/java/org/openrewrite/kotlin/KotlinParser.java

```diff
@@ -380,153 +348,98 @@ public KotlinParser.Builder clone() {
         }
     }
 
-    public CompiledSource parse(List<Parser.Input> sources, Disposable disposable, ExecutionContext ctx) {
-        CompilerConfiguration compilerConfiguration = compilerConfiguration();
-        if (classpath != null) {
-            for (Path path : classpath) {
-                File file;
-                try {
-                    file = path.toFile();
-                } catch (UnsupportedOperationException ex) {
-                    continue;
-                }
-                addJvmClasspathRoot(compilerConfiguration, file);
+    // TODO: Compiler Source File args need to be an actual file path therefore copying synthetic file to temp folder
+    public static String tempFile(ExecutionContext ctx, Input source, AtomicInteger idx) {
+        String fileName;
+        if ("openRewriteFile.kt".equals(source.getPath().toString())) {
+            fileName = "openRewriteFile" + idx.getAndIncrement() + ".kt";
+        } else if ("openRewriteFile.kts".equals(source.getPath().toString())) {
+            fileName = "openRewriteFile" + idx.getAndIncrement() + ".kts";
+        } else {
+            fileName = source.getPath().toString();
+        }
+        if(TEMP_SOURCE_DIR == null) {
+            try {
+                TEMP_SOURCE_DIR = Files.createTempDirectory("rewrite");
+            } catch (IOException e) {
+                throw new RuntimeException(e);
             }
         }
-        addJvmClasspathRoot(compilerConfiguration, PathUtil.getResourcePathForClass(AnnotationTarget.class));
+        Path tempFile = TEMP_SOURCE_DIR.resolve(fileName);
+        try {
+            Files.createDirectories(tempFile.getParent());
+            Files.copy(source.getSource(ctx), tempFile, StandardCopyOption.REPLACE_EXISTING);
+        } catch (FileAlreadyExistsException ignored) {
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        return tempFile.toAbsolutePath().toString();
+    }
```

Comment:
Calling this out for my colleagues as creating files on disk might be a little different when it concerns the CLI / workers.